### PR TITLE
Add our own hex::serde implementation

### PR DESCRIPTION
**Pull request by mxz42**
part of #1307 

the serialize and deserialize functions are specific to `Vec<u8>` but I think that works so far. 

thoughts? is it ok like this or should be more generic? 
@dpc @elsirion

### CODE REVIEW

File: fedimint-api/src/hex/serde.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+use std::borrow::Cow;
+
+use bitcoin_hashes::hex::{FromHex, ToHex};
+use serde::de::Error;
+use serde::{Deserialize, Serializer};
+
+/// Serialize a `Vec<u8>` to a hex String
+pub fn serialize<S>(data: &Vec<u8>, serializer: S) -> Result<S::Ok, S::Error>
```
[the remark this answers is not in the document]
Comment (mxz42) [reply fedimint-api/src/hex/serde.rs:8]: oh, yes, of course. thanks.

File: fedimint-api/src/hex/serde.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+where
+    S: Serializer,
+{
+    let hex: String = data.to_hex();
+    serializer.serialize_str(&hex)
+}
+
+/// Deserialize a hex String to a `Vec<u8>`
+pub fn deserialize<'de, D: serde::Deserializer<'de>>(d: D) -> Result<Vec<u8>, D::Error> {
+    let hex_bytes: Cow<'de, str> = Deserialize::deserialize(d)?;
```
Comment (mxz42) [inline fedimint-api/src/hex/serde.rs:18]: I thought about this lifetime here for `Cow<'de, str>`. It works with and without `'de`. I think it's fine to attach the lifetime of the Cow to the deserializer here?

File: fedimint-api/src/hex/serde.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+use std::borrow::Cow;
+
+use bitcoin_hashes::hex::{FromHex, ToHex};
+use serde::de::Error;
+use serde::{Deserialize, Serializer};
+
+/// Serialize a `Vec<u8>` to a hex String
+pub fn serialize<S>(data: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+where
```
Comment (elsirion) [inline fedimint-api/src/hex/serde.rs:9]: > Serialize a `Vec<u8>` to a hex String

Is this a documentation bug now?

**Review by elsirion — DISMISSED** [review]
One minor nit

File: fedimint-api/src/hex/serde.rs (hunk 3 above)
Comment (mxz42) [reply fedimint-api/src/hex/serde.rs:9]: yes it is! thanks

**Comment by mxz42** [comment]
should be ready to get merged @elsirion

**Review by elsirion — APPROVED** [review]

**Review by dpc — APPROVED** [review]

**Comment by dpc** [comment]
@mxz42 Please upstream it and become (if you're not already) rust-bitcoin contributor.

**Pull request merged**